python/frontend/cirrus/cirrus/core.py

```diff
@@ -118,17 +127,17 @@ def get_updates_per_second(self, fetch=True):
             t = time.time() - self.start_time
             ups = messenger.get_num_updates(self.ps_ip_public, self.ps_ip_port)
             self.time_ups_lst.append((t, ups))
+
+            cost_per_second = self.cost_model.get_cost(t)
+            if ups is not None and not (ups == 0):
```

Comment:
Why not ```ups != 0```?

Reply:
ups is None when messenger.get_num_update errors. However the second clause in the condition is probably better as ups != 0. Fixed